### DISCUSSION TITLE
Make private view components public

### DISCUSSION THREAD

**Discussion by severinkilchhofer** [directus/directus#13139] [Archive]
Hi, We are in the middle of writing a custom block for the Editor.js interface plugin with the goal to let the user select a relation of a given collection. For that we would like to use the drawer-collection component which exists in the source code but isn’t registered in https://github.com/directus/directus/blob/main/app/src/components/register.ts to be publicly reusable. Is there a reason for that? Everything seems to work when we copy/paste this component to the block plugin and use as local component.

**Comment by benhaynes** [comment]
I'd defer to @rijkvanzanten on that one, but I know we have to be careful about exposing some internals since then changes we make to those could be considered breaking changes (to extensions) which limits our ability to efficiently develop the platform. But again, I might be wrong about that!

**Reply by rijkvanzanten** [reply]
That's 100% it

**Reply by severinkilchhofer** [reply]
Thank you for your clarification. I can understand this decision.

**Reply by hamwithglasses** [reply]
@rijkvanzanten @benhaynes Sorry for commenting on an old topic and if this is a dumb question. But related to this drawer topic, would using something like editItem() function within the list-m2m interface essentially trigger the drawer for that given item?

**Reply by hanneskuettner** [reply]
Heya! Same etiquette as on Discord. Please do not tag any maintainers of this project. We are very aware of every comment made and will respond if time permits. Tagging people does not make it more likely to receive an answer.

Also no need to ask questions in a somewhat unrelated discussion if you don't receive an answer to a question asked on Discord within a few hours. 

Thank you! 🙏
